src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -309,6 +309,7 @@ public virtual TContext InitializeGlobalContextFromOptions(TOptions options, ref
             context.PluginFilePaths = options.PluginFilePaths?.Any() == true ? options.PluginFilePaths?.ToImmutableHashSet() : context.PluginFilePaths;
             context.InsertProperties = options.InsertProperties?.Any() == true ? InitializeStringSet(options.InsertProperties) : context.InsertProperties;
             context.MaxFileSizeInKilobytes = options.MaxFileSizeInKilobytes != null ? options.MaxFileSizeInKilobytes.Value : context.MaxFileSizeInKilobytes;
+            context.TimeoutInMilliseconds = options.TimeoutInMilliseconds != null ? Math.Max(options.TimeoutInMilliseconds.Value, 0) : context.TimeoutInMilliseconds;
```

Comment:
The TimeoutInMilliseonds in option will override the setting in context, which will be applied to Run 
<img width="814" alt="image" src="https://github.com/microsoft/sarif-sdk/assets/119978670/08e22a09-1d4d-4343-88b6-ae76b2a226f6">